rdflib/plugins/sparql/parser.py

```diff
@@ -227,8 +227,17 @@ def expandCollection(terms):
 )
 
 
+percent_encoded_reserved_chars_re = re.compile(
+    """%2[0-13-9a-cfA-CF]|%3[abdfABDF]|%40|%5[bdBD]"""
+)
+
+
 def _hexExpand(match):
```

Comment:
I was thinking about this a bit - should we even be doing `_hexExpand` at all on URIs?

Reply:
Will do some testing on it.

Reply:
I tested this now, works fine if I just remove all of this:

https://github.com/RDFLib/rdflib/blob/ba855c233310afee65abd67f2f77239f909cd895/rdflib/plugins/sparql/parser.py#L230-L234

And I'm not sure if there is a good reason to not remove it.

Will make a commit to your PR to remove this and also move the tests to `test/test_sparql/test_prefixed_name.py` together with the addition of some more tests.

Reply:
np.

Reply:
Done now.